File: 2 CPP/Codeforces-2014A-RobinHelps.cpp
#include <iostream>
using namespace std;

int main() {
    
    int nc;
    cin >> nc;
    
    for(int i=0; i<nc;i++){
        int n,k;
        cin >> n >> k ;
        int stock = 0, ans = 0;
        
        for(int j=0; j<n; j++){
            int gold;
            cin >> gold;
            if(gold==0 and stock>0){
                stock-=1;
                ans +=1 ;
            };
            
            if (gold>=k){
                stock+=gold;
            };
        };
        
        cout << ans << endl;
    };
    
    return 0;
}

File: 1 Python/Codeforces-2014A-RobinHelps.py
nc = int(input())

for _ in range(nc):
    n,k = map(int, input().rstrip().split())
    gArr = list(map(int, input().rstrip().split()))
    stock = 0
    ans = 0
    for i in range(n):
        if gArr[i]==0 and stock>0:
            stock-=1
            ans+=1
        if gArr[i]>=k:
            stock+= gArr[i]
    
    print(ans)
